### PR TITLE
Deneb spec alignment

### PR DESCRIPTION
**Pull request by olegshmuelov**
Integration of Deneb Fork Compatibility Features

### CODE REVIEW

File: beacon/goclient/validator.go [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain validators: %w", err)
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("validators response is nil")
+	}
+	if resp.Data == nil {
+		return nil, fmt.Errorf("validators data is nil")
+	}
```
Comment (moshe-blox) [inline beacon/goclient/validator.go:25]: Remove nil check to match previous behavior?

File: protocol/v2/ssv/runner/proposer.go [hunk 2]
```diff
@@ -425,34 +428,63 @@ type blockSummary struct {
...
 		switch b.Version {
 		case spec.DataVersionCapella:
 			return summarizeBlock(b.Capella)
+		case spec.DataVersionDeneb:
+			if b.Deneb == nil {
+				return summary, fmt.Errorf("deneb block contents is nil")
+			}
+			if b.Deneb.Block == nil {
+				return summary, fmt.Errorf("deneb block is nil")
+			}
```
Comment (moshe-blox) [inline protocol/v2/ssv/runner/proposer.go:453]: Unnecessary because its checked in summarizeBlock

**Review by moshe-blox — CHANGES_REQUESTED** [review]

File: beacon/goclient/attest.go [hunk 3]
```diff
@@ -14,20 +16,42 @@ import (
...
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain attester duties: %w", err)
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("attester duties response is nil")
+	}
+	if resp.Data == nil {
+		return nil, fmt.Errorf("attester duties data is nil")
+	}
```
[the remark this answers is not in the document]
Comment (y0sher) [reply beacon/goclient/attest.go:31]: @moshe-blox wouldn't `Data` still be full as long as go-eth2-client is on the new version?

File: beacon/goclient/goclient.go [hunk 4]
```diff
@@ -174,11 +170,15 @@ func New(logger *zap.Logger, opt beaconprotocol.Options, operatorID spectypes.Op
...
+	nodeVersionResp, err := client.client.NodeVersion(opt.Context, &api.NodeVersionOpts{})
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to get node version")
+		return nil, fmt.Errorf("failed to get node version: %w", err)
+	}
+	if nodeVersionResp == nil {
+		return nil, fmt.Errorf("node version response is nil")
 	}
-	client.nodeClient = ParseNodeClient(client.nodeVersion)
+	client.nodeVersion = nodeVersionResp.Data
```
Comment (y0sher) [inline beacon/goclient/goclient.go:180]: at other places you check if `Data` is `nil` why not here?

File: beacon/goclient/proposer.go [hunk 5]
```diff
@@ -28,160 +27,215 @@ const (
...
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain proposer duties: %w", err)
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("proposer duties response is nil")
+	}
+	if resp.Data == nil {
+		return nil, fmt.Errorf("proposer duties data is nil")
+	}
```
[the remark this answers is not in the document]
Comment (y0sher) [reply beacon/goclient/proposer.go:42]: @moshe-blox it seems to me we have to leave this check as long as we're using the new version? like the go-eth2-client will use this format even for before-fork.

File: protocol/v2/ssv/spectest/ssv_mapping_test.go [hunk 6]
```diff
@@ -97,10 +97,10 @@ func prepareTest(t *testing.T, logger *zap.Logger, name string, test interface{}
 				typedTest.Run(t)
 			},
 		}
-	case reflect.TypeOf(&messages.MsgSpecTest{}).String(): // no use of internal structs so can run as spec test runs
+	case reflect.TypeOf(&partialsigmessage.MsgSpecTest{}).String(): // no use of internal structs so can run as spec test runs
```
Comment (y0sher) [inline protocol/v2/ssv/spectest/ssv_mapping_test.go:100]: dont we need both?

File: beacon/goclient/goclient.go (hunk 4 above)
Comment (olegshmuelov) [reply beacon/goclient/goclient.go:180]: the Data is string in this case

File: integration/qbft/tests/test_duty.go [hunk 7]
```diff
@@ -31,7 +31,7 @@ func createDuty(pk []byte, slot phase0.Slot, idx phase0.ValidatorIndex, role spe
 	case spectypes.BNRoleAggregator:
 		testingDuty = spectestingutils.TestingAggregatorDuty
 	case spectypes.BNRoleProposer:
-		testingDuty = *spectestingutils.TestingProposerDutyV(spec.DataVersionBellatrix)
+		testingDuty = *spectestingutils.TestingProposerDutyV(spec.DataVersionCapella)
```
[the remark this answers is not in the document]
Comment (olegshmuelov) [reply integration/qbft/tests/test_duty.go:34]: those tests are not used

File: protocol/v2/ssv/spectest/ssv_mapping_test.go (hunk 6 above)
Comment (olegshmuelov) [reply protocol/v2/ssv/spectest/ssv_mapping_test.go:100]: this package changed in spec

File: protocol/v2/ssv/testing/ssv_msgs.go [hunk 8]
```diff
@@ -308,8 +301,8 @@ var postConsensusBeaconBlockMsg = func(
 	blsSig := spec.BLSSignature{}
 	copy(blsSig[:], sig)
 
-	signed := capella.SignedBeaconBlock{
-		Message:   testingutils.TestingBeaconBlockCapella,
+	signed := &deneb.SignedBeaconBlock{
```
[the remark this answers is not in the document]
Comment (olegshmuelov) [reply protocol/v2/ssv/testing/ssv_msgs.go:304]: I'll revert this change
btw those tests in ssv_msgs are not used at all, prob all of them

**Pull request closed without merging**